### PR TITLE
feat(lambda): add SnapStart support for arm64 functions

### LINKED ISSUE #30899: ❗️NOTICE (lambda): SnapStart for Java functions using ARM64
### Describe the feature

The new feature is currently being worked on in [this PR](https://github.com/aws/aws-cdk/pull/30898). In the meantime, you can access the SnapStart feature using the following workaround:

```typescript
    const fn = new lambda.Function(this, 'Func', {
      runtime: lambda.Runtime.JAVA_21,
      handler: 'example.Handler',
      architecture: lambda.Architecture.ARM_64,
      code: lambda.Code.fromAsset(path.join(__dirname, '../../lambda/example.jar')),
      memorySize: 256,
    });

    // Workaround: Add SnapStart configuration like this:
    (fn.node.defaultChild as lambda.CfnFunction).addPropertyOverride('SnapStart', {
      ApplyOn: 'PublishedVersions',
    });
```

### CDK version used

2.149.0

### Environment details (OS name and version, etc.)

All

### PR DESCRIPTION
**Pull request by valerena**
### Issue # (if applicable)

Closes #30899.

### Reason for this change

Lambda released SnapStart support for arm64 functions: https://aws.amazon.com/about-aws/whats-new/2024/07/aws-lambda-snapstart-java-functions-arm64-architecture/

### Description of changes

Allow configuring SnapStart on arm64 functions.

### Description of how you validated changes

```yarn test```

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by valerena** [comment]
In the meantime while this gets merged and officially released, people wanting to use SnapStart on their arm64 functions, need to use the workaround that was used before SnapStart was supported in the function L2 construct:
(based on https://github.com/aws/aws-cdk/issues/23153#issuecomment-1331196914)

```ts
    const fn = new lambda.Function(this, 'Func', {
      runtime: lambda.Runtime.JAVA_21,
      handler: 'example.Handler',
      architecture: lambda.Architecture.ARM_64,
      code: lambda.Code.fromAsset(path.join(__dirname, '../../lambda/example.jar')),
      memorySize: 256,
    });

    // Workaround: Add SnapStart configuration like this:
    (fn.node.defaultChild as lambda.CfnFunction).addPropertyOverride('SnapStart', {
      ApplyOn: 'PublishedVersions',
    });
```

File: packages/aws-cdk-lib/aws-lambda/README.md [hunk 1]
```diff
@@ -967,7 +967,7 @@ managing concurrency.
 
 ## Lambda with SnapStart
 
-SnapStart is currently supported only on Java 11/Java 17 runtime. SnapStart does not support provisioned concurrency, the arm64 architecture, Amazon Elastic File System (Amazon EFS), or ephemeral storage greater than 512 MB. After you enable Lambda SnapStart for a particular Lambda function, publishing a new version of the function will trigger an optimization process.
+SnapStart is currently supported only on Java 11/Java 17 runtime. SnapStart does not support provisioned concurrency, Amazon Elastic File System (Amazon EFS), or ephemeral storage greater than 512 MB. After you enable Lambda SnapStart for a particular Lambda function, publishing a new version of the function will trigger an optimization process.
```
Comment (msailes) [inline packages/aws-cdk-lib/aws-lambda/README.md:970]: SnapStart is also available on Java 21. 

You might want to flip this around to say "SnapStart is not available on Java 8 (AL1 or AL2)" or words to that effect.

Comment (valerena) [reply packages/aws-cdk-lib/aws-lambda/README.md:970]: We need to mention that it's only supported **in Java**, so we can't leave "just Java8" out of the support.  (Yes, @msailes, there are other runtimes other than Java 😉)

The [official documentation](https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html) mentions "SnapStart supports Java 11 and later Java managed runtimes", so we can go with something like that.

**Review by xazhao — DISMISSED** [review]
LGTM!

**Review by Leo10Gama — APPROVED** [review]

**Pull request merged**